src/main/java/jenkins/bouncycastle/api/PEMManager.java

```diff
@@ -0,0 +1,394 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, CloudBees, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package jenkins.bouncycastle.api;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.security.DigestInputStream;
+import java.security.Key;
+import java.security.KeyPair;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.cert.Certificate;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import org.apache.commons.codec.binary.Hex;
+import org.apache.commons.io.FileUtils;
+import org.bouncycastle.openssl.PEMReader;
+import org.bouncycastle.openssl.PEMWriter;
+import org.bouncycastle.openssl.PasswordException;
+import org.bouncycastle.openssl.PasswordFinder;
+import org.bouncycastle.util.encoders.Base64;
+
+/**
+ * A class that provides an API to manager PEM format, providing additional methods to handle Keys, Certificates,
+ * Fingerprints, etc The supported algorithms will depend on the underlying version of BouncyCastle
+ *
+ */
+public class PEMManager {
+
+    /**
+     * Stores the internal Bouncy Castle or JCA object
+     */
+    private Object pemObject;
+
+    /**
+     * Creates a {@link PEMManager} by reading a PEM file
+     * 
+     * @param pemFile {@link File} pointing to the PEM file to read
+     * @throws IOException launched if a problem exists reading the PEM information or the {@link File}
+     * @throws BCPasswordException in case PEM is passphrase protected and none or wrong is provided
```

Comment:
The critical test is can the caller be expected to recover from the exception.

If the caller cannot be expected to recover -> unchecked

If the caller can be expected to recover -> checked

If in doubt -> unchecked

In this case we are not in doubt. This is the same class of error as an IOException and we are assuming Java 7 for most users of this (as Java 7 is 1.612 or something like that) so multi-catch blocks should not be an issue.

Plus it is an existing exception that exactly matches the use case